mediorum/server/helpers.go

```diff
@@ -20,14 +22,24 @@ func apiPath(parts ...string) string {
 	return u.String()
 }
 
-// replaces the host portion of a URL, maintaining path and query params
-func replaceHost(u url.URL, newHost string) url.URL {
+// replaces the host portion of a URL, maintaining path and query params, and setting the scheme to https for prod/stage
+func (ss *MediorumServer) replaceHost(c echo.Context, newHost string) url.URL {
+	u := *c.Request().URL
 	u.Host = cleanHost(newHost)
+	u.Scheme = ss.getScheme()
```

Comment:
can't we leave existing scheme untouched?

Reply:
the reason I did this was because `dest.String()` was returning a relative URL, which worked on the client. but when trying to parse it out from the golang side I was running into some issue which was fixed by making it relative. I forget which part exactly errored